File: frontend/src/services/forumReplyService.js
// =============================================================
// forumReplyService.js — Forum Reply API Calls
// =============================================================
// Handles HTTP requests for creating and fetching replies on forum posts.
// Communicates with Spring Boot ForumReplyController at /api/forum-replies.
//
// API Flow:
//   createReply        → POST /api/forum-replies          → Save a reply to a specific post
//   getRepliesByPostId → GET  /api/forum-replies/{postId} → Fetch all replies for one post
// =============================================================

import API from "./api"

// -------------------------
// Create a reply to a forum post
// -------------------------
// Called when a user submits a reply under a specific discussion post.
// Payload includes: postId (which post is being replied to), author, content.
export const createReply =
  async (replyData) => {

    const response =
      await API.post(
        "/forum-replies",
        replyData
      )

    return response.data
}

// -------------------------
// Get all replies for a specific post
// -------------------------
// Fetches all replies belonging to a given postId.
// Used in Forum.jsx to display threaded discussions under each post.
export const getRepliesByPostId =
  async (postId) => {

    const response =
      await API.get(
        `/forum-replies/${postId}`
      )

    return response.data
}

File: backend/src/main/java/com/farmersbuddy/farmers_buddy_backend/service/ForumReplyService.java
package com.farmersbuddy.farmers_buddy_backend.service;

import com.farmersbuddy.farmers_buddy_backend.entity.ForumReply;

import com.farmersbuddy.farmers_buddy_backend.repository.ForumReplyRepository;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

// =============================================================
// ForumReplyService.java — Business Logic Layer for Forum Replies
// =============================================================
// Handles business logic for creating and retrieving replies to forum posts.
// Acts as the intermediary between ForumReplyController and ForumReplyRepository.
//
// Architecture Position:
//   ForumReplyController → ForumReplyService → ForumReplyRepository → MySQL
//
// Replies are linked to their parent post via the postId field,
// and fetched using the findByPostId() derived query in the repository.
// =============================================================

@Service
public class ForumReplyService {

    // Injected by Spring — provides access to the forum_replies table in MySQL
    @Autowired
    private ForumReplyRepository forumReplyRepository;

    // -------------------------
    // Create a New Forum Reply
    // -------------------------
    // Saves a reply to a specific forum post and returns the saved entity.
    // The forumReply object must include postId, author, and content.
    public ForumReply createReply(
            ForumReply forumReply
    ) {

        return forumReplyRepository.save(
                forumReply
        );
    }

    // -------------------------
    // Get Replies by Post ID
    // -------------------------
    // Fetches all replies belonging to a specific forum post.
    // Delegates to the custom findByPostId() method in the repository,
    // which Spring Data JPA resolves to: SELECT * FROM forum_replies WHERE post_id = ?
    public List<ForumReply> getRepliesByPostId(
            Long postId
    ) {

        return forumReplyRepository.findByPostId(
                postId
        );
    }

    // -------------------------
    // Delete a Forum Reply
    // -------------------------
    public void deleteReply(Long id) {
        forumReplyRepository.deleteById(id);
    }
}
